modules/actor/actor-wfrp4e.js

```diff
@@ -1790,11 +1790,14 @@ export default class ActorWfrp4e extends WFRP4eDocumentMixin(Actor)
   }
 
 
-  applyTerror(value, name = undefined) {
+  async applyTerror(value, name = undefined) {
     value = value || 1
     let terror = duplicate(game.wfrp4e.config.systemItems.terror)
     terror.flags.wfrp4e.terrorValue = value
-    return game.wfrp4e.utility.applyOneTimeEffect(terror, this)
+    let scripts = new EffectWfrp4e(terror, {parent: this}).getScripts();
```

Comment:
This class doesn't have a `getScripts` function, like actors and items. You can simply call the `scripts` getter on an effect and it will construct all the script objects.

Reply:
changed